src/network/bmproto.py

```diff
@@ -438,17 +483,20 @@ def bm_command_version(self):
         logger.debug("my external IP: %s", self.sockNode.host)
         logger.debug("remote node incoming address: %s:%i", self.destination.host, self.peerNode.port)
         logger.debug("user agent: %s", self.userAgent)
-        logger.debug("streams: [%s]", ",".join(map(str,self.streams)))
+        logger.debug("streams: [%s]", ",".join(map(str, self.streams)))
         if not self.peerValidityChecks():
-            # TODO ABORT
+            # .. todo:: ABORT
```

Comment:
I think this isn't necessary. Returning here will prevent verack being sent. Without that the handshake is considered incomplete, other commands are ignored and the connection will timeout after 20 seconds. The error in the write buffer has this time to make it through. Probably a leftover from early stages, please remove this "todo".